lec05-dplyr.Rmd

```diff
@@ -1,4 +1,510 @@
 ---
-title: "More on using dplyr and a brief introduction to plotting with ggplot"
+title: "Tidying and exporting data, reproducing figures"
 ---
 
+> ### Lecture objectives
+> 
+> - Learn about tidy data.
+> - Transform data from the long to wide format.
+> - Reproduce existing figures from raw data.
+> - Understand which raw data is underlying a figure.
+> - Understand which types of figures are suitable to create from raw data.
+> - Explore scientific questions using dplyr and ggplot.
+> 
+> 
+> ### Lecture outline
+> 
+> - Reshaping with gather and spread (25 min)
+> - Exporting data (15 min)
+> - Reproducing figures (50 min)
+
+
+## Setting up
+
+Start by loading the required packages. Both **`ggplot2`** and **`dplyr`** are
+included in the **`tidyverse`** package collection.
+
+```{r}
+# Install if needed
+# install.packages('tidyverse')
+library(tidyverse)
+```
+
+Load the data we saved in the previous lesson.
+
+```{r}
+# Download if needed
+# download.file("https://ndownloader.figshare.com/files/2292169", "data/portal_data.csv")
+surveys <- read_csv('portal_data.csv')
+surveys
+```
+
+## Reshaping with gather and spread
+
+**`dplyr`** is one part of a larger **`tidyverse`** that enables you to work
+with data in tidy data formats. **`tidyr`** enables a wide range of
+manipulations of the structure data itself. For example, the survey data
+presented here is in almost in what we call a **long** format - every
+observation of every individual is its own row. This is an ideal format for data
+with a rich set of information per observation. It makes it difficult, however,
+to look at the relationships between measurements across plots. For example,
+what is the relationship between mean weights of different genera across the
+entire data set?
+
+To answer that question, we'd want each plot to have a single row, with all of
```

Comment:
`plot` being land plot, right? This name is really quite unfortunate...

Reply:
True, should have changed it in the firs lecture with this dataset I guess...